### PR TITLE
feat(report) Add RolesGuard and Rename functions

### PR DESCRIPTION
**Pull request by theoatboy139**

### CODE REVIEW

File: src/report/report.service.ts [hunk 1]
```diff
@@ -14,19 +14,19 @@ export class ReportService {
...
     }
 
-    getUserReports(): Promise<Report[]> {
+    findUserReports(): Promise<Report[]> {
         return this.model.find({ type: ReportType.User }).exec();
     }
 
-    getSystemReports(): Promise<Report[]> {
+    findSystemReports(): Promise<Report[]> {
         return this.model.find({ type: ReportType.System }).exec();
```
Comment (UtopiaBeam) [inline src/report/report.service.ts]: You can remove duplicated `find()`, `findSystemReports()` and `findUserSystems()` by accept a `filters` in `find` like this:

```ts
find(filters): Promise<Report[]> {
    return this.model.find(filters).exec();
}
```
And then just pass these filters from controller.

File: src/report/report.service.ts [hunk 2]
```diff
@@ -14,22 +14,14 @@ export class ReportService {
         private readonly userService: UserService,
     ) {}
 
-    getAllReports(): Promise<Report[]> {
-        return this.model.find().exec();
+    find(filters): Promise<Report[]> {
+        return this.model.find(filters).exec();
```
Comment (UtopiaBeam) [inline src/report/report.service.ts]: Sorry, I forgot that `filter` should be optional.

File: src/report/report.controller.ts [hunk 3]
```diff
@@ -4,34 +4,43 @@ import { ReportService } from './report.service';
...
+        return this.service.find({ type: ReportType.User });
     }
 
+    @Roles(UserRole.Admin)
     @Get('system')
-    getSystemReports() {
-        return this.service.getSystemReports();
+    findSystemReports() {
+        return this.service.find({ type: ReportType.User });
     }
```
Comment (DarkXenoZ) [inline src/report/report.controller.ts]: is this supposed to be ReportType.System?

Comment (theoatboy139) [reply src/report/report.controller.ts]: Yes

**Pull request merged**